src/nativeMain/kotlin/types/EclairApiTypes.kt

```diff
@@ -28,4 +29,130 @@ data class NodeInfo(
+data class DisconnectionResult(val success: Boolean, val message: String) : EclairApiType()
+
+@Serializable
+data class Node(
+    val signature: String,
+    val features: Features,
+    val timestamp: Timestamp,
+    val nodeId: String,
+    val rgbColor: String,
+    val alias: String,
+    val addresses: List<String>,
+    val tlvStream: Map<String, @Contextual Any>
+)
+
+@Serializable
+data class Timestamp(
+    val iso: String,
+    val unix: Long
+)
+
+@Serializable
+data class NodeCommandResponse(
+    val announcement: Node
+) : EclairApiType()
+
+@Serializable
+data class AllChannels(
+    val shortChannelId: String,
+    val a: String,
+    val b: String
+)
+
+@Serializable
+data class AllUpdates(
+    val signature: String,
+    val chainHash: String,
+    val shortChannelId: String,
+    val timestamp: Timestamp,
+    val messageFlags: Map<String, Boolean>,
+    val channelFlags: ChannelFlags,
+    val cltvExpiryDelta: Int,
+    val htlcMinimumMsat: Int,
+    val feeBaseMsat: Int,
+    val feeProportionalMillionths: Int,
+    val htlcMaximumMsat: Long,
+    val tlvStream: Map<String, @Contextual Any>
+)
+
+@Serializable
+data class ChannelFlags(
+    val isEnabled: Boolean,
+    val isNode1: Boolean
+)
+
+@Serializable
+data class Invoice(
+    val prefix: String,
+    val timestamp: Long,
+    val nodeId: String,
+    val serialized: String,
+    val description: String,
+    val paymentHash: String,
+    val paymentMetadata: String,
+    val expiry: Int,
+    val minFinalCltvExpiry: Int,
+    val amount: Long? = null,
+    val features: Features,
+    val routingInfo: List<@Contextual Any>
+) : EclairApiType()
+
+@Serializable
+data class DeleteInvoiceResult(val success: Boolean, val message: String) : EclairApiType()
+
+@Serializable
+data class InvoiceResult(val success: Boolean, val message: String) : EclairApiType()
+
+@Serializable
+data class RouteResult(
+    val paymentId: String,
+    val parentId: String
+) : EclairApiType()
+
+@Serializable
+data class SentInfoResponse(
+    val id: String,
+    val parentId: String,
+    val paymentHash: String,
+    val paymentType: String,
+    val amount: Long,
+    val recipientAmount: Long,
+    val recipientNodeId: String,
+    val createdAt: Timestamp,
+    val invoice: Invoice,
+    val status: PaymentStatus
+)
+
+@Serializable
+data class PaymentStatus(
+    val type: String,
+    val paymentPreimage: String,
+    val feesPaid: Long,
+    val route: List<Route>,
+    val completedAt: Timestamp
+)
+
+@Serializable
+data class Route(
+    val nodeId: String,
+    val nextNodeId: String,
+    val shortChannelId: String
+)
+
+@Serializable
+data class ReceivedInfoResponse(
+    val invoice: Invoice,
+    val paymentPreimage: String,
+    val paymentType: String,
+    val createdAt: Timestamp,
+    val status: ReceivePaymentStatus
+) : EclairApiType()
+
+@Serializable
+data class ReceivePaymentStatus(
+    val type: String,
+    val amount: Long? = null,
```

Comment:
Do you really need to define a default value? It would be better if we didn't.

Reply:
> Do you really need to define a default value? It would be better if we didn't.

From what I have understood so far, when I have a nullable field in my data class `val amount: Long?` without a default value and the JSON that I'm trying to deserialize does not contain that field, the kotlinx-serializtion throws an exception because it expects a value but doesn't find it.
If I just provide a default value `val amount: Long? = null`, now the field is missing in the JSON, but still it could be deserialized now.
Also, if I do not provide a default value, then a few test cases fail because of the reason I mentioned. For example, here's a log which explains it.
```
> Task :cleanNativeTest
> Task :compileKotlinNative
> Task :nativeProcessResources NO-SOURCE
> Task :nativeMainKlibrary
> Task :compileTestKotlinNative
> Task :linkDebugTestNative
> Task :nativeTest FAILED

kotlin.AssertionError: Expected value to be null, but was: <ApiError(code=1, message=api response could not be parsed: [
  {
    "prefix": "lnbcrt",
    "timestamp": 1643719798,
    "nodeId": "028e2403fbfddb3d787843361f91adbda64c6f622921b19fb48f5766508bcadb29",
    "serialized": "lnbcrt1psljtrkpp5gqus3ys83p9cry4nj43ykjyvkuuhrcc5y45a6l569zwuc8pn2xxsdq0ydex2cmtd3jhxucsp543t76xycc9kpx4estwm6tjlpsht3m7d5jxe09tqnyjjux970y9lsmqz9gxqrrsscqp79qtzsqqqqqysgqxwh55ncvj3hv0cypm8vafku83gayzg7qa3zlu3lua76lk53t2m3rgt4d5qa04cfdd0f407p328c9el9xvy3r6z9um90m5pjaxrrazysqfkxfa7",
    "description": "#reckless",
    "paymentHash": "4039089207884b8192b395624b488cb73971e3142569dd7e9a289dcc1c33518d",
    "paymentMetadata": "2a",
    "expiry": 3600,
    "minFinalCltvExpiry": 30,
    "features": {
      "activated": {
        "payment_secret": "mandatory",
        "basic_mpp": "optional",
        "option_payment_metadata": "optional",
        "var_onion_optin": "mandatory"
      },
      "unknown": []
    },
    "routingInfo": []
  },
  {
    "prefix": "lnbcrt",
    "timestamp": 1643719828,
    "nodeId": "028e2403fbfddb3d787843361f91adbda64c6f622921b19fb48f5766508bcadb29",
    "serialized": "lnbcrt1psljty5pp5z2247c5w8cl30err7s9qx8rkejltq49mk8z6l5eqar7l43pehapsdq0ydex2cmtd3jhxucsp5hrcu5s0jftrmje4yavu580tlrq4mdmdevye8aevn6dsae4x5kejqmqz9gxqrrsscqp79qtzsqqqqqysgqwus3au5085tp02cwvpjexc5rq6qjezwxwr3yecdxr525qprv5zjxa98r69kx87cegavjw9u9299yfdhnes7mp4dztttyduchudvq64cq4pyx28",
    "description": "#reckless",
    "paymentHash": "12955f628e3e3f17e463f40a031c76ccbeb054bbb1c5afd320e8fdfac439bf43",
    "paymentMetadata": "2a",
    "expiry": 3600,
    "minFinalCltvExpiry": 30,
    "features": {
      "activated": {
        "payment_secret": "mandatory",
        "basic_mpp": "optional",
        "option_payment_metadata": "optional",
        "var_onion_optin": "mandatory"
      },
      "unknown": []
    },
    "routingInfo": []
  }
])>.
	at kotlin.Throwable#<init>(Unknown Source)
	at kotlin.Error#<init>(Unknown Source)
	at kotlin.AssertionError#<init>(Unknown Source)
	at kotlin.test.DefaultAsserter#fail(Unknown Source)
	at kotlin.test.Asserter#assertTrue(Unknown Source)
	at kotlin.test.Asserter#assertNull(Unknown Source)
	at kotlin.test#assertNull(Unknown Source)
	at kotlin.test#assertNull$default(Unknown Source)
	at commands.ListPendingInvoicesCommandTest#successful(Unknown Source)
	at commands.$ListPendingInvoicesCommandTest$test$0.$successful(Unknown Source)
	at commands.$ListPendingInvoicesCommandTest$test$0.$successful(Unknown Source)
	at kotlin.native.internal.test.BaseClassSuite.TestCase#run(Unknown Source)
	at kotlin.native.internal.test.TestRunner.run#internal(Unknown Source)
	at kotlin.native.internal.test.TestRunner.runIteration#internal(Unknown Source)
	at kotlin.native.internal.test.TestRunner#run(Unknown Source)
	at kotlin.native.internal.test#testLauncherEntryPoint(Unknown Source)
	at kotlin.native.internal.test#main(Unknown Source)
	at <global>.Konan_start(Unknown Source)
	at <global>.Init_and_run_start(Unknown Source)
	at <global>.0x0(Unknown Source)
	at <global>.__libc_start_main(Unknown Source)
	at <global>.0x0(Unknown Source)



commands.ListPendingInvoicesCommandTest.successful request FAILED
    kotlin.AssertionError at null:-1
1 test completed, 1 failed
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':nativeTest'.
> There were failing tests. See the report at: file:///home/claddy/Documents/eclair-cli/build/reports/tests/nativeTest/index.html
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 9s
5 actionable tasks: 5 executed
```

Reply:
Interesting, it's a pity that kotlinx-serialization isn't smarter, in that case we should keep the default value of `null`.